exchanges/okx/okx.go

```diff
@@ -1060,149 +739,201 @@ func (ok *Okx) GetOneClickRepayHistory(ctx context.Context, after, before time.T
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getTradeAccountRateLimitEPL, http.MethodGet, "trade/account-rate-limit", nil, &resp, request.AuthenticatedRequest)
+}
+
+// OrderPreCheck precheck the account information before and after placing the order.
+// Only applicable to Multi-currency margin mode, and Portfolio margin mode
+func (ok *Okx) OrderPreCheck(ctx context.Context, arg *OrderPreCheckParams) (*OrderPreCheckResponse, error) {
+	if arg == nil {
+		return nil, common.ErrNilPointer
+	}
+	if arg.InstrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	if arg.TradeMode == "" {
+		return nil, errInvalidTradeModeValue
+	}
+	if arg.Side == "" {
+		return nil, order.ErrSideIsInvalid
+	}
+	if arg.OrderType == "" {
+		return nil, order.ErrTypeIsInvalid
+	}
+	if arg.Size <= 0 {
+		return nil, order.ErrAmountBelowMin
+	}
+	var resp *OrderPreCheckResponse
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, orderPreCheckEPL, http.MethodPost, "trade/order-precheck", arg, &resp, request.AuthenticatedRequest)
 }
 
 /*************************************** Block trading ********************************/
 
-// GetCounterparties retrieves the list of counterparties that the user has permissions to trade with.
+// GetCounterparties retrieves the list of counterparties that the user has permissions to trade with
 func (ok *Okx) GetCounterparties(ctx context.Context) ([]CounterpartiesResponse, error) {
 	var resp []CounterpartiesResponse
-	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getCounterpartiesEPL, http.MethodGet, rfqCounterparties, nil, &resp, true)
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getCounterpartiesEPL, http.MethodGet, "rfq/counterparties", nil, &resp, request.AuthenticatedRequest)
 }
 
-// CreateRfq Creates a new Rfq
-func (ok *Okx) CreateRfq(ctx context.Context, arg CreateRfqInput) (*RfqResponse, error) {
+// CreateRFQ Creates a new RFQ
+func (ok *Okx) CreateRFQ(ctx context.Context, arg *CreateRFQInput) (*RFQResponse, error) {
 	if len(arg.CounterParties) == 0 {
 		return nil, errInvalidCounterParties
 	}
 	if len(arg.Legs) == 0 {
-		return nil, errInvalidLegs
-	}
-	var resp []RfqResponse
-	err := ok.SendHTTPRequest(ctx, exchange.RestSpot, createRfqEPL, http.MethodPost, rfqCreateRfq, &arg, &resp, true)
-	if err != nil {
-		return nil, err
-	}
-	if len(resp) == 1 {
-		return &resp[0], nil
+		return nil, errMissingLegs
 	}
-	return nil, errNoValidResponseFromServer
+	var resp *RFQResponse
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, createRFQEPL, http.MethodPost, "rfq/create-rfq", &arg, &resp, request.AuthenticatedRequest)
 }
 
-// CancelRfq Cancel an existing active Rfq that you has previously created.
-func (ok *Okx) CancelRfq(ctx context.Context, arg CancelRfqRequestParam) (*CancelRfqResponse, error) {
-	if arg.RfqID == "" && arg.ClientRfqID == "" {
-		return nil, errMissingRfqIDAndClientRfqID
+// CancelRFQ Cancel an existing active RFQ that you has previously created
```

Comment:
* `has` => `have`
* Implied terms: existing, active and previously created, so we can remove them all
```suggestion
// CancelRFQ cancels an Request for quotation
```